content/6_Advanced/Parallel_Binary_Search.mdx

```diff
@@ -0,0 +1,142 @@
+---
+id: parallel-binary-search
+title: 'Parallel Binary Search'
+author: Muaath Alqarni
+description: 'Answer binary search queries offline'
+prerequisites:
+  - binary-search
+  - 'Divide & Conquer'
+frequency: 0
+---
+
+
+## Resources
+
+<Resources>
+	<Resource source="CF" title="Parallel Binary Seach [tutorial]" url="45578" />
+	<Resource source="SPBook" title="Parallel Binary Seach" url="https://kostka.dev/sp/spbook.pdf#page=22" />
+</Resources>
+
+## Tutorial
+This technique is usually used when we can solve a single query by doing linear or binary search on monotonic function.
+
+The trick is doing it offline, sort queries by their median (initially $l$, $r$ will be the binary search values), do linear search,
+whenever you find a query its median in same as your position in linear search, check the monotonic function, and do same as binary search (edit either $l$ or $r$).
```

Comment:
If that's what you meant, then resolve those changes. If not, then please rewrite that sentence since it doesn't really make sense right now.